test/unit/coroutine/stress.cpp

```diff
@@ -0,0 +1,62 @@
+#include <util/time.hpp>
+
+#include <yaclib/async/run.hpp>
+#include <yaclib/coroutine/await.hpp>
+#include <yaclib/coroutine/future_coro_traits.hpp>
+#include <yaclib/executor/thread_pool.hpp>
+#include <yaclib/fault/atomic.hpp>
+#include <yaclib/fault/thread.hpp>
+
+#include <exception>
+#include <iostream>
+#include <random>
+#include <stack>
+#include <utility>
+#include <vector>
+
+#include <gtest/gtest.h>
+
+namespace {
+using namespace std::chrono_literals;
+using namespace yaclib;
+
+yaclib_std::atomic_size_t state = 0;
+
+Future<size_t> incr(uint64_t delta) {
+  size_t old = state.fetch_add(delta, std::memory_order_acq_rel);
+  co_return old;
+}
+
+TEST(StressCoro, IncAtomic) {
+  constexpr size_t kTestCases = 100'000;
+  constexpr size_t kAwaitedFutures = 20;
+  constexpr size_t kMaxDelta = 10;
+
+  auto tp = MakeThreadPool();
```

Comment:
tp not used, so you don't need atomic. (You also forgot Stop)

I also think we need two version of this test, one single-threaded, another multi.